exchange/bitswap/bitswap.go

```diff
@@ -308,154 +253,76 @@ func (bs *Bitswap) sendWantlistToProviders(ctx context.Context, entries []wantli
 			defer cancel()
 			providers := bs.network.FindProvidersAsync(child, k, maxProvidersPerRequest)
 			for prov := range providers {
-				sendToPeers <- prov
+				go func(p peer.ID) {
+					bs.network.ConnectTo(ctx, p)
+				}(prov)
 			}
 		}(e.Key)
 	}
 
-	go func() {
-		wg.Wait() // make sure all our children do finish.
-		close(sendToPeers)
-	}()
-
-	err := bs.sendWantlistToPeers(ctx, sendToPeers)
-	if err != nil {
-		log.Debugf("sendWantlistToPeers error: %s", err)
-	}
+	wg.Wait() // make sure all our children do finish.
```

Comment:
i don't think this will respect the context. this needs:

```go
done := make(chan struct{})   
go func() {   
  wg.Wait()   
  close(done)   
}()   
  
select {    
case <-done:    
case <-ctx.Done():    
  // NB: we may be abandoning goroutines here before they complete    
  // this shouldnt be an issue because they will complete soon anyways    
  // we just don't want their being slow to impact bitswap transfer speeds    
}   
return nil
```